tests/auth_tests/test_views.py

```diff
@@ -253,7 +253,23 @@ def test_confirm_invalid_post(self):
         })
         # Check the password has not been changed
         u = User.objects.get(email='staffmember@example.com')
-        self.assertTrue(not u.check_password("anewpassword"))
+        self.assertFalse(u.check_password("anewpassword"))
+
+    def test_confirm_invalid_hash(self):
+        # Same as test_confirm_invalid, but trying to do a POST with an invalid hash instead.
+        u = User.objects.get(email='staffmember@example.com')
+        url, path = self._test_confirm_start()
+        path_parts = path.split('-')
+        path_parts[-1] = ("0") * 20 + '/'
+        path = '-'.join(path_parts)
+
+        self.client.post(path, {
+            'new_password1': 'anewpassword',
+            'new_password2': 'anewpassword',
+        })
+        # Check the password has not been changed
```

Comment:
I'd rather be explicit here and refresh from the database. On that note, there's `u.refresh_from_db()` :wink: